adapters/http-vertx/src/main/java/org/eclipse/hono/adapter/http/vertx/VertxBasedHttpProtocolAdapter.java

```diff
@@ -85,27 +83,9 @@ public void setClientCertAuthProvider(final HonoClientBasedAuthProvider provider
         this.clientCertAuthProvider = Objects.requireNonNull(provider);
     }
 
-    /**
-     * Gets the options for configuring the server side trust anchor.
-     * <p>
-     * This implementation returns the options returned by
-     * {@link AbstractConfig#getTrustOptions()} if not {@code null}.
-     * Otherwise, it returns trust options for verifying a client
-     * certificate's validity period.
-     * 
-     * @return The trust options.
-     */
     @Override
     protected TrustOptions getServerTrustOptions() {
-
-        return Optional.ofNullable(getConfig().getTrustOptions())
-                .orElseGet(() -> {
-                    if (getConfig().isAuthenticationRequired()) {
-                        return new ValidityBasedTrustOptions();
-                    } else {
-                        return null;
-                    }
-                });
+        return getTrustOptions();
```

Comment:
I do not see why we need to introduce another level of indirection. Can't we simply pull up this method *as is*?

Reply:
if you say "pull up", you mean having it as a default implementation for all protocol adapters?

Reply:
Don't you think that would be a reasonable default?